common/client/node_lifecycle.go

```diff
@@ -151,47 +153,51 @@ func (n *node[CHAIN_ID, HEAD, RPC]) aliveLoop() {
 
 	for {
 		select {
-		case <-n.nodeCtx.Done():
+		case <-ctx.Done():
 			return
 		case <-pollCh:
-			var version string
-			promPoolRPCNodePolls.WithLabelValues(n.chainID.String(), n.name).Inc()
-			lggr.Tracew("Polling for version", "nodeState", n.State(), "pollFailures", pollFailures)
-			ctx, cancel := context.WithTimeout(n.nodeCtx, pollInterval)
-			version, err := n.RPC().ClientVersion(ctx)
-			cancel()
-			if err != nil {
-				// prevent overflow
-				if pollFailures < math.MaxUint32 {
-					promPoolRPCNodePollsFailed.WithLabelValues(n.chainID.String(), n.name).Inc()
-					pollFailures++
+			if !func(ctx context.Context) bool {
```

Comment:
Why the huge anonymous function?

Reply:
To be able to properly defer the `cancel()` call. Do you think it is worth declaring a new method instead?

Reply:
Would creating a derived context/cancelFn and canceling the new cancelFn help avoid the anon function?

Reply:
Can't defer it if we do that :shrug:

Reply:
Yeah much cleaner